lib/test-collection.js

```diff
@@ -64,7 +66,15 @@ TestCollection.prototype.add = function (test) {
 			throw new Error('"only" cannot be used with a ' + type + ' test');
 		}
 
-		this.hooks[type].push(test);
+		// add a always hook
+		if (metadata.always) {
+			if (type !== 'after' && type !== 'afterEach') {
+				throw new Error('"always" cannot be used with a ' + type + ' test');
+			}
+			this.hooks[type + 'Always'].push(test);
```

Comment:
Also, a few tests in `test/test-collection.js` verifying new tests are added correctly (this will involve a modification to the `serialize` function in that test).